`device_next/usbd_ch9.c` depends on `CONFIG_HWINFO=y`, but `CONFIG_HWINFO` is only implied by `CONFIG_USB_DEVICE_STACK_NEXT`

### Describe the bug

As in the title - the user can n-select `CONFIG_HWINFO`, because it is only implied by `CONFIG_USB_DEVICE_STACK_NEXT`. The change from `select` to `imply` happened in: 9b8796400a7a5e440ce2f90ab3d0aff51c0f8f9e.



### Regression

- [x] This is a regression.

### Steps to reproduce

To reproduce, use the following commands:
1.  `west build -p -b stm32f429i_disc1/stm32f429xx samples/subsys/usb/shell -- -DCONFIG_HWINFO=n`
2. See output

### Relevant log output

```shell
/home/filip/Work/zephyr-maintenance/zephyr/subsys/usb/device_next/usbd_ch9.c:536:32: error: 'CONFIG_USBD_HWINFO_DEVID_LENGT
' undeclared here (not in a function)
  536 | #define USBD_SN_ASCII7_LENGTH (CONFIG_USBD_HWINFO_DEVID_LENGTH * 2)
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/filip/Work/zephyr-maintenance/zephyr/subsys/usb/device_next/usbd_ch9.c:536:32: note: in definition of macro 'USBD_SN_ASCII7_LENGTH'
  536 | #define USBD_SN_ASCII7_LENGTH (CONFIG_USBD_HWINFO_DEVID_LENGTH * 2)
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/filip/Work/zephyr-maintenance/zephyr/subsys/usb/device_next/usbd_ch9.c: In function 'get_sn_from_hwid':
/home/filip/Work/zephyr-maintenance/zephyr/subsys/usb/device_next/usbd_ch9.c:536:32: error: 'CONFIG_USBD_HWINFO_DEVID_LENGT
' undeclared (first use in this function)
  536 | #define USBD_SN_ASCII7_LENGTH (CONFIG_USBD_HWINFO_DEVID_LENGTH * 2)
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/filip/Work/zephyr-maintenance/zephyr/subsys/usb/device_next/usbd_ch9.c:536:32: note: in definition of macro 'USBD_SN_ASCII7_LENGTH'
  536 | #define USBD_SN_ASCII7_LENGTH (CONFIG_USBD_HWINFO_DEVID_LENGTH * 2)
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/filip/Work/zephyr-maintenance/zephyr/subsys/usb/device_next/usbd_ch9.c:536:32: note: each undeclared identifier is reported only once for each function it appears in
  536 | #define USBD_SN_ASCII7_LENGTH (CONFIG_USBD_HWINFO_DEVID_LENGTH * 2)
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/filip/Work/zephyr-maintenance/zephyr/subsys/usb/device_next/usbd_ch9.c:536:32: note: in definition of macro 'USBD_SN_ASCII7_LENGTH'
  536 | #define USBD_SN_ASCII7_LENGTH (CONFIG_USBD_HWINFO_DEVID_LENGTH * 2)
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/filip/Work/zephyr-maintenance/zephyr/subsys/usb/device_next/usbd_ch9.c:542:17: warning: unused variable 'hwid' [-Wunused-variable]
  542 |         uint8_t hwid[USBD_SN_ASCII7_LENGTH / 2U];
      |                 ^~~~
/home/filip/Work/zephyr-maintenance/zephyr/subsys/usb/device_next/usbd_ch9.c: In function 'string_ascii7_to_utf16le':
/home/filip/Work/zephyr-maintenance/zephyr/subsys/usb/device_next/usbd_ch9.c:536:32: error: 'CONFIG_USBD_HWINFO_DEVID_LENGT
' undeclared (first use in this function)
  536 | #define USBD_SN_ASCII7_LENGTH (CONFIG_USBD_HWINFO_DEVID_LENGTH * 2)
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/filip/Work/zephyr-maintenance/zephyr/subsys/usb/device_next/usbd_ch9.c:536:32: note: in definition of macro 'USBD_SN_ASCII7_LENGTH'
  536 | #define USBD_SN_ASCII7_LENGTH (CONFIG_USBD_HWINFO_DEVID_LENGTH * 2)
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/filip/Work/zephyr-maintenance/zephyr/subsys/usb/device_next/usbd_ch9.c:579:62: error: type of formal parameter 1 is incomplete
  579 |                 ssize_t sn_ascii7_str_len = get_sn_from_hwid(sn_ascii7_str);
      |                                                              ^~~~~~~~~~~~~
/home/filip/Work/zephyr-maintenance/zephyr/subsys/usb/device_next/usbd_ch9.c:569:17: warning: unused variable 'sn_ascii7_st
' [-Wunused-variable]
  569 |         uint8_t sn_ascii7_str[USBD_SN_ASCII7_LENGTH];
      |                 ^~~~~~~~~~~~~
[183/201] Building C object zephyr/kernel/CMakeFiles/kernel.dir/sched.c.obj
ninja: build stopped: subcommand failed.
```

### Impact

Showstopper – Prevents release or major functionality; system unusable.

### Environment

- OS: Linux
- Toolchain: Zephyr SDK
- Commit SHA: 31ef45e2f11543a07773fd0e802efe2409da9638

### Additional Context

_No response_

--- a/subsys/usb/device_next/Kconfig
+++ b/subsys/usb/device_next/Kconfig
@@ -96,8 +96,7 @@ config USBD_MSG_WORK_DELAY
 	  yet ready to publish the message. The delay unit is milliseconds.
 
 config USBD_HWINFO_DEVID_LENGTH
-	int "The length of the device ID requested from HWINFO in bytes"
-	depends on HWINFO
+	int "The length of the device ID requested from HWINFO in bytes" if HWINFO
 	range 8 128
 	default 16
 	help

